RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -1,112 +1,311 @@
+  private boolean autoClimbPressed = false; // is the autoclimb button currently being pressed
 
-  private WPI_TalonSRX climber;
-  private boolean fixedHookToggleState;
-  private boolean elevatorToggle;
-  private boolean autoClimb;
+  private ClimbState currentClimbState;
   DoubleSolenoid elevatorSolenoid;
-  DoubleSolenoid fixedHookSolenoid;
+  
+  private int fullExtendPosition = 5000; // TODO: change this number
+  private int partialExtendPosition = fullExtendPosition / 2;
+  private int fullRetractPosition = 0; // TODO: change this number
+
+  // https://docs.ctre-phoenix.com/en/stable/ch16_ClosedLoop.html#mechanism-is-finished-command
+  private int climbErrThreshold1;
+  private int climbLoopsToSettle1;
+  private int withinThresholdLoops1;
+
+  private int climbErrThreshold2;
+  private int climbLoopsToSettle2;
+  private int withinThresholdLoops2;
+
+  private boolean climberTilted = false;
 
   private final Changeable<Double> climbOutput = BucketLog.changeable(Put.DOUBLE, "climber/climbOutput", 0.5);
 
   private final Loggable<String> climbState = BucketLog.loggable(Put.STRING, "climber/climbState");
-  private final Loggable<String> hookToggleState = BucketLog.loggable(Put.STRING, "climber/hookState");
-  private final Loggable<String> elevatorToggleState = BucketLog.loggable(Put.STRING, "climber/elevatorState");
+  private final Loggable<String> elevatorTiltedState = BucketLog.loggable(Put.STRING, "climber/elevatorTiltedState");
+  private final Loggable<Double> climberMotorPosition = BucketLog.loggable(Put.DOUBLE, "climber/climberMotorPosition");
 
   public ClimberSubsystem(Config config) {
     super(config);
   }
 
   @Override
   public void init() {
-    climber = new WPI_TalonSRX(config.climberMotor_ID);
+    climber1 = MotorUtils.makeSRX(config.climber.climber1);
+    climber2 = MotorUtils.makeSRX(config.climber.climber2);
+
     if (config.enablePneumatics) {
-      elevatorSolenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH, config.
-      elevatorSolenoid_ID1, config.elevatorSolenoid_ID2);
-      fixedHookSolenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH, config.fixedHookSolenoid_ID1, config.fixedHookSolenoid_ID2);
+      elevatorSolenoid =
+        new DoubleSolenoid(PneumaticsModuleType.REVPH, config.elevatorSolenoid_ID1, config.elevatorSolenoid_ID2);
     }
+
+    currentClimbState = ClimbState.Idle;
+  }
+
+  boolean isClimberAtSetpoint()
+  {
+    climbErrThreshold1 = 10;
+    climbLoopsToSettle1 = 10;
+    withinThresholdLoops1 = 0;
+    boolean climber1AtSetpoint = false;
+    if (climber1.getClosedLoopError() < +climbErrThreshold1 && climber1.getClosedLoopError() > -climbErrThreshold1) {
+      withinThresholdLoops1++;
+
+      if (withinThresholdLoops1 > climbLoopsToSettle1)
+        climber1AtSetpoint = true;
+    } else {
+      withinThresholdLoops1 = 0;
+    }
+    
+    climbErrThreshold2 = 10;
+    climbLoopsToSettle2 = 10;
+    withinThresholdLoops2 = 0;
```

Comment:
When testing in the sim, I noticed this resets the `withinThresholdLoops2` to 0 during each check, so we never get within the threshold for 10 loops. Whoops! :)